lua/soulbound_abilities/discombob.lua

```diff
@@ -0,0 +1,134 @@
+local ABILITY = {}
+
+ABILITY.Name = "Discombobulator"
+ABILITY.Id = "discombob"
+ABILITY.Description = "Throw a discombobulator"
+ABILITY.Icon = "vgui/ttt/icon_nades"
+
+local discombob_fuse_time = CreateConVar("ttt_soulbound_discombob_fuse_time", "5", FCVAR_NONE, "How long the fuse for the Soulbound's discombobulator should be", 0, 10)
+local discombob_uses = CreateConVar("ttt_soulbound_discombob_uses", "2", FCVAR_REPLICATED, "How many uses should of the discombobulator ability should the Soulbound get. (Set to 0 for unlimited uses)", 0, 10)
+local discombob_cooldown = CreateConVar("ttt_soulbound_discombob_cooldown", "1", FCVAR_NONE, "How long should the Soulbound have to wait between uses of the discombobulator ability", 0, 10)
+
+table.insert(ROLE_CONVARS[ROLE_SOULBOUND], {
+    cvar = "ttt_soulbound_discombob_fuse_time",
+    type = ROLE_CONVAR_TYPE_NUM,
+    decimal = 1
+})
+table.insert(ROLE_CONVARS[ROLE_SOULBOUND], {
+    cvar = "ttt_soulbound_discombob_uses",
+    type = ROLE_CONVAR_TYPE_NUM,
+    decimal = 0
+})
+table.insert(ROLE_CONVARS[ROLE_SOULBOUND], {
+    cvar = "ttt_soulbound_discombob_cooldown",
+    type = ROLE_CONVAR_TYPE_NUM,
+    decimal = 1
+})
+
+function ABILITY:Bought(soulbound)
+    soulbound:SetNWInt("TTTSoulboundDiscombobUses", discombob_uses:GetInt())
+    soulbound:SetNWFloat("TTTSoulboundDiscombobNextUse", CurTime())
+end
+
+function ABILITY:Condition(soulbound, target)
+    if not soulbound:IsInWorld() then return false end
+    if discombob_uses:GetInt() > 0 and soulbound:GetNWInt("TTTSoulboundDiscombobUses", 0) <= 0 then return false end
+    if CurTime() < soulbound:GetNWFloat("TTTSoulboundDiscombobNextUse") then return false end
+    return true
+end
+
+function ABILITY:Use(soulbound, target)
+    local ang = soulbound:EyeAngles()
+    local src = soulbound:GetPos() + (soulbound:Crouching() and soulbound:GetViewOffsetDucked() or soulbound:GetViewOffset()) + (ang:Forward() * 8) + (ang:Right() * 10)
+    local pos = soulbound:GetEyeTraceNoCursor().HitPos
+    local tang = (pos-src):Angle() -- A target angle to actually throw the grenade to the crosshair instead of fowards
+    -- Makes the grenade go upgwards
+    if tang.p < 90 then
+        tang.p = -10 + tang.p * ((90 + 10) / 90)
+    else
+        tang.p = 360 - tang.p
+        tang.p = -10 + tang.p * -((90 + 10) / 90)
+    end
+    tang.p = math.Clamp(tang.p, -90, 90) -- Makes the grenade not go backwards :/
```

Comment:
The HL2 mod I used to work on had a bug where on Linux servers (which we never released), all thrown grenades would go toward the origin 0,0,0 of the map, resulting in many backwards throws...